traits/observers/parsing.py

```diff
@@ -0,0 +1,262 @@
+# (C) Copyright 2005-2020 Enthought, Inc., Austin, TX
+# All rights reserved.
+#
+# This software is provided without warranty under the terms of the BSD
+# license included in LICENSE.txt and may be redistributed only under
+# the conditions described in the aforementioned license. The license
+# is also available online at http://www.enthought.com/licenses/BSD.txt
+#
+# Thanks for using Enthought open source!
+
+import contextlib
+from functools import reduce
+import operator
+
+import traits.observers.expression as _expr_module
+from traits.observers._generated_parser import (
+    Lark_StandAlone as _Lark_StandAlone
+)
+
+_LARK_PARSER = _Lark_StandAlone()
+
+#: Token annotation for a name (a trait name, or a metadata name, etc.)
+_NAME_TOKEN = "NAME"
```

Comment:
I this something we should be importing from somewhere? Where/how does Lark store the string constant it uses for token types?

Reply:
This `"NAME"` is data from the grammar, and so it is going to be somewhere in the generated code `DATA` or `MEMO` variable. However those two structures are not really for human consumption.

On the other hand, this `_NAME_TOKEN` is here only for sanity check. I can remove it too and just not check it.

Reply:
It's a bit odd if Lark exposes `token.type` as part of the public API but doesn't expose the constants for `token.type` to be compared to. How are consumers supposed to use `token.type`?

Reply:
Sorry for not being clear...
I meant the `"NAME"` is defined by us:
https://github.com/enthought/traits/blob/db76bc8e070d8ca63276ad4087b560190e14f205/traits/observers/_dsl_grammar.lark#L51

If I change the grammar file such that `"NAME: /[a-zA-Z_]\w*/"` becomes `"NAME2: /[a-zA-Z_]\w*/"` (and everywhere else that uses it), then regenerate the standalone parser. This `_NAME_TOKEN = "NAME"` will need to change to `_NAME_TOKEN = "NAME2"`.

Reply:
If it is okay, I am going to remove this sanity check. It is on the border of being not really necessary given we already have tests for the parsed outcomes.

Reply:
The sanity check for `"NAME"` is not covered in tests. Likewise, this line is not covered either.
```
            raise RuntimeError("Default notify flag unexpectedly changed.") 
```